addons/im_livechat/static/src/core/common/chat_window_model_patch.js

```diff
@@ -13,30 +13,34 @@ const chatWindowPatch = {
         super.setup(...arguments);
         this.livechatStep = CW_LIVECHAT_STEP.NONE;
     },
-    close(...args) {
+    close(options = {}) {
         if (this.thread?.channel_type !== "livechat") {
-            return super.close(...args);
+            return super.close(...arguments);
+        }
+        if (options.force) {
+            this.livechatStep = CW_LIVECHAT_STEP.NONE;
+            super.close(...arguments);
         }
         const isSelfVisitor = this.thread.livechatVisitorMember?.persona?.eq(this.store.self);
         switch (this.livechatStep) {
             case CW_LIVECHAT_STEP.NONE: {
                 if (this.thread.isTransient) {
                     this.thread.delete();
-                    super.close(...args);
+                    super.close(...arguments);
                     break;
                 }
                 if (!this.thread.livechat_active) {
                     if (isSelfVisitor) {
                         this.livechatStep = CW_LIVECHAT_STEP.FEEDBACK;
                         this.open({ notifyState: this.thread?.state !== "open" });
                     } else {
-                        super.close(...args);
+                        super.close(...arguments);
                     }
                     break;
                 }
                 this.actionsDisabled = true;
                 this.livechatStep = CW_LIVECHAT_STEP.CONFIRM_CLOSE;
-                if (!this.hubAsOpened) {
+                if (!this.hubAsOpened && options.notifyState) {
                     this.open();
                 }
                 break;
```

Comment:
Why this `options.notifyState` here? For the close I get (this is the main change of this PR), but here it auto-opens to ask for confirmation. If the chat is folded in this step, it must auto-open to ask for confirmation.

Reply:
FYI the PR in question, merged quite recently: https://github.com/odoo/odoo/pull/194402